ocrd_webapi/authentication.py

```diff
@@ -0,0 +1,52 @@
+from hashlib import sha512
+from random import random
+from typing import Tuple
+
+from .database import create_user, get_user
+
+
+async def authenticate_user(email: str, password: str):
```

Comment:
IIUC then currently it is possible to just add a user and use it to authenticate because `UserAccountDB.validated_user` is currently not taken into account when authenticating users.

Reply:
Any user could just register and then do anything. So if you just start the webapi it seems like you have authentication but in reality anybody could do anything. If you want to use it with working authentication now (as I need for the deployment in core) you would have to change the code to get authentication. Previously it was looking bad but it was working at least, now that is taken away

Reply:
As I said, that's not taken into account yet. But it will be before the PR is merged. Nothing is going to break due to this merge. The default account will still function like an admin account for the user that deploys the WebAPI. I opened it for review to get more ideas on how to improve or to find out if I'm doing something wrong.

Reply:
This is improved now and implemented as it should be. I have refactored `validated_user` to `approved_user` to differentiate and avoid potential confusion between credential validation and user approval of an admin. Have a look.